### PR TITLE
Add API products

### PR DESCRIPTION
**Pull request by gionn**
- [X] basic introduction
- [x] explain rate limiter

### CODE REVIEW

File: docs/api-product.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Secure access to the API is managed by the platform, generating an **unique
+token** during the provisioning process for each consumer, that expires when the
+subscription is terminated.
+
+For this reason, the API URL is never directly exposed to the consumer. Instead,
+an intermediate endpoint is automatically managed by the platform to let the
+consumer reach the API backend is a secure and controlled way.
+
+When a consumer send a request to the API, an additional set of HTTP headers is
```
Comment (AndreaVecchi) [inline docs/api-product.md]: "sends"

File: docs/api-product.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+added to the request arriving at the API backend:
+
+* `X-Consumer-ID`: an auto-generated ID of the Consumer
+* `X-Consumer-Custom-ID`: the id of the user that bought the subscription
```
Comment (AndreaVecchi) [inline docs/api-product.md]: which

File: docs/api-product.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* `X-Consumer-Username`: the username of the user that bought the subscription
+
+## Pricing models
+
+In addition to the [standard pricing
+options](onboarding.md#available-pricing-models) available on every plan, it's
+possible to define a prepaid pricing model based on the actual HTTP calls
```
Comment (AndreaVecchi) [inline docs/api-product.md]: current

File: docs/api-product.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+invoked by the consumer.
+
+### Prepaid HTTP call pricing
+
+For every HTTP call that needs to be billed, create a resource with:
```
Comment (AndreaVecchi) [inline docs/api-product.md]: link to "extra resources"

File: docs/api-product.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+id: api-product
+title: API products onboarding
+sidebar_label: API products
+---
+
+In this section it is presented how to onboard and start selling an API-based
```
Comment (malteo) [inline docs/api-product.md]: This section describes how...

File: docs/api-product.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Catalog, as described [here](onboarding.md#applications-catalogue).
+
+For extended integration purposes, it's possible to define a syndication
+endpoint like in [syndicated products](syndication.md) to receive events about
+the creation and termination of the related subscriptions for a specific
+product.
+
+## Authentication
+
+Secure access to the API is managed by the platform, generating an **unique
```
Comment (malteo) [inline docs/api-product.md]: a unique

http://www.writersdigest.com/online-editor/a-before-consonants-and-an-before-vowels-is-not-the-rule

File: docs/api-product.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+For this reason, the API URL is never directly exposed to the consumer. Instead,
+an intermediate endpoint is automatically managed by the platform to let the
+consumer reach the API backend is a secure and controlled way.
+
+When a consumer sends a request to the API, an additional set of HTTP headers is
+added to the request arriving at the API backend:
+
+* `X-Consumer-ID`: an auto-generated ID of the Consumer
+* `X-Consumer-Custom-ID`: the id of the user which bought the subscription
```
Comment (malteo) [inline docs/api-product.md]: vogliamo specificare che è l'id del customer della *nostra* piattaforma?

**Review by malteo — CHANGES_REQUESTED** [review]

**Review by malteo — APPROVED** [review]

**Pull request merged**